src/core/linter-rules/check-punctuation.js

```diff
@@ -0,0 +1,45 @@
+/**
+ * Linter rule "check-punctuation". Makes sure the there are no punctuations missing at the end of a <p>
+ *   in the ReSpec config.
+ */
+import { lang as defaultLang } from "core/l10n";
+import LinterRule from "core/LinterRule";
+
+const name = "check-punctuation";
+const punctuationMarks = [".", ":", "!", "?"];
+
+const meta = {
+  en: {
+    description: "`<p>` tags should end with either of ., :, !, ? ",
+    howToFix: "Please append a ., :, !, ? at the end of this `<p>` tag",
+  },
+};
+// Fall back to english, if language is missing
+const lang = defaultLang in meta ? defaultLang : "en";
+
+/**
+ * Runs linter rule.
+ *
+ * @param {Object} config The ReSpec config.
+ * @param  {Document} doc The document to be checked.
+ */
+function lintingFunction(conf, doc) {
+  
+  // ensures that either a string ends with one of [.!?:] or is empty
+  const punctuatingRegExp = new RegExp(`[${punctuationMarks.join("")}]$|^ *$`, "m");
+  
+  const offendingElements = Array.from(doc.querySelectorAll("p"))
+    .filter((elem) => !(punctuatingRegExp.test(elem.textContent)));
+  
+  offendingElements.splice(-1,1); // back-to-top is always the last element.
```

Comment:
We want to filter this out via css (i.e., via “:not()”. This is probably causing the issue you mentioned below.

Reply:
As a general rule, and even though not applicable here, never use `splice()`, alway `filter()`. You should treat all data sources (in this case “offendingElements”) as immutable.

Reply:
See https://github.com/w3c/respec/pull/1589#comment-376754462

Reply:
Thanks! I've pushed the new changes

Reply:
@marcoscaceres look up the new commit whenever u are free